DataPlotly/gui/plot_settings_widget.py

```diff
@@ -483,7 +483,11 @@ def getJSmessage(self, status):
             # if a clicking event is performed depending on the plot type
             elif dic["mode"] == 'clicking':
                 if dic['type'] == 'scatter':
-                    self.layer_combo.currentLayer().selectByIds([dic['fidd']])
+                    exp = """ {} = '{}' """.format(dic['field'], dic['id'])
```

Comment:
mmmm.. ok then we can stick with this patch. What do you think?

Reply:
Why do you want to stick to the current patch ?

You can make your code easier : 
L484, L486 and L495 by using `selectByExpression` instead of `selectByIds` and having to create a loop.
`getFeatures` is not very efficient, even if you use a QgsFeatureRequest if you don't specify just the fields you need (ID in your example). You can read the blogpost https://nyalldawson.net/2016/10/speeding-up-your-pyqgis-scripts/